webservice/tscanservice/tscanwrapper.py

```diff
@@ -101,40 +104,40 @@ def get_tree(filename, index):
-            sentence, filename, index = line.split("\t")
-            if filename.startswith("input/"):
-                filename = filename.replace("input/", inputdir)
-            yield sentence, get_tree(filename, int(index))
+def get_output_trees(inputdir, lines: List[str]):
+    for line in lines:
+        sentence, filename, index = line.split("\t")
+        if filename.startswith("input/"):
+            filename = filename.replace("input/", inputdir)
+        yield sentence, get_tree(filename, int(index))
 
 
 def init_alpino_lookup(configfile, inputdir, outputdir):
-    alpino_lookup = []
-
-    # hidden files containing the output of the parsing are added to
-    # the input folder, to speed up (re)analyzing the same sentences
-    alpino_filenames = set(
-        chain(
-            (alpino_xml.filename for alpino_xml in clamdata.inputfiles("alpino")),
-            (os.path.basename(x) for x in glob.iglob(inputdir + ".*.alpino.xml"))))
-
-    for alpino_filename in alpino_filenames:
-        filepath = inputdir + alpino_filename
-        corpus = etree.parse(filepath)
-        root = corpus.getroot()
-        # treebanks start counting from zero, xpath queries are also 1-based
-        trees = [(0, root)] if root.tag == 'alpino_ds' else enumerate(
-            corpus.findall("alpino_ds"), 1)
-        for index, tree in trees:
-            # reconstruct the sentence from the words
-            words = {}
-            for word in tree.findall(".//node[@word]"):
-                words[int(word.attrib['begin'])] = word.attrib['word']
-            sentence = ' '.join(
-                map(lambda x: x[1], sorted(words.items(), key=lambda x: x[0])))
-            alpino_lookup.append((sentence, filepath, index))
+    def yield_lookup():
+        # hidden files containing the output of the parsing are added to
+        # the input folder, to speed up (re)analyzing the same sentences
+        alpino_filenames = set(
+            chain(
+                (alpino_xml.filename for alpino_xml in clamdata.inputfiles("alpino")),
+                (os.path.basename(x) for x in glob.iglob(inputdir + ".*.alpino.xml"))))
+
+        for alpino_filename in alpino_filenames:
+            filepath = inputdir + alpino_filename
+            corpus = etree.parse(filepath)
+            root = corpus.getroot()
+            # treebanks start counting from zero, xpath queries are also 1-based
+            trees = [(0, root)] if root.tag == 'alpino_ds' else enumerate(
+                corpus.findall("alpino_ds"), 1)
+            for index, tree in trees:
+                # reconstruct the sentence from the words
+                words = {}
+                for word in tree.findall(".//node[@word]"):
+                    words[int(word.attrib['begin'])] = word.attrib['word']
+                sentence = ' '.join(
+                    map(lambda x: x[1], sorted(words.items(), key=lambda x: x[0])))
+                yield (sentence, filepath, index)
```

Comment:
Please deforest and maybe take a loot at standard `operator.{attrgetter,methodcaller,itemgetter}`. `yield_lookup` can be a separate function if you just pass `inputdir`.